File: mobile/src/main/java/net/maiatoday/levelbest/model/Entry.java
package net.maiatoday.levelbest.model;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.util.Date;

/**
 * POJO to define basic mood/energy tracking datapoint
 * Created by maia on 2016/07/27.
 */

public class Entry {

    public static final int MOOD = 0;
    public static final int ENERGY_LEVEL = 1;
    public static final int ENERGY_TYPE = 2;


    @IntDef({MOOD, ENERGY_LEVEL, ENERGY_TYPE})
    @Retention(RetentionPolicy.SOURCE)
    public @interface EntryType {
    }

    int id;
    long dateTime;
    @EntryType
    int entryType;
    float level;
    String comment;

    public Entry(int id, int entryType, float level) {
        this.id = id;
        this.entryType = entryType;
        this.level = level;
        this.comment = "";
        this.dateTime = new Date().getTime();

    }

    public Entry(int id, long dateTime, int entryType, float level) {
        this.id = id;
        this.dateTime = dateTime;
        this.entryType = entryType;
        this.level = level;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public
    @EntryType
    int getEntryType() {
        return entryType;
    }

    public void setEntryType(@EntryType int entryType) {
        this.entryType = entryType;
    }

    public float getLevel() {
        return level;
    }

    public void setLevel(float level) {
        this.level = level;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public long getDateTime() {
        return dateTime;
    }

    public void setDateTime(long dateTime) {
        this.dateTime = dateTime;
    }

    @Override
    public String toString() {
        return "Entry{" +
                "id=" + id +
                ", dateTime='" + dateTime + '\'' +
                ", entryType=" + entryTypeToString(entryType) + "(" + entryType + ")" +
                ", level=" + level +
                ", comment='" + comment + '\'' +
                '}';
    }

    public static String entryTypeToString(@EntryType int et) {
        switch (et) {
            case MOOD:
                return "Mood";
            case ENERGY_LEVEL:
                return "Energy Level";
            case ENERGY_TYPE:
                return "Energy Type";
            default:
                return "Unknown";
        }
    }
}

File: mobile/src/main/kotlin/net/maiatoday/levelbest/model/Entry.kt
package net.maiatoday.levelbest.model

import java.util.Date

import io.realm.RealmList
import io.realm.RealmModel
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import io.realm.annotations.Required

/**
 * Mood/Energy data point entry
 * Created by maia on 2017/02/04.
 */
@RealmClass
open class Entry : RealmModel {
    @Required @PrimaryKey open var id: String = ""
    open var timeStamp: Date = Date()
    open var lat: Float = 0.0F
    open var long: Float = 0.0F
    open var tags: RealmList<Tag> = RealmList()
    open var moods: RealmList<Mood> = RealmList()
    open var sentiment: Int = 0
    open var energyLevel: Int = 0
    open var note:String = ""
    open var title:String = ""

    companion object {
        val TIMESTAMP = "timeStamp"

    }
}
